custom/icds/models.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.app_manager.dbaccessors import get_build_by_version
+from corehq.motech.utils import b64_aes_decrypt
+from custom.icds.utils.ccz_hosting import CCZHostingUtility
+from custom.icds.validators import (
+    LowercaseAlphanumbericValidator,
+)
+
+
+class CCZHostingLink(models.Model):
+    identifier = models.CharField(null=False, unique=True, max_length=255, db_index=True,
+                                  validators=[LowercaseAlphanumbericValidator])
+    username = models.CharField(null=False, max_length=255)
+    # b64_aes_encrypt'ed raw password is stored in DB
+    password = models.CharField(null=False, max_length=255)
+    domain = models.CharField(null=False, max_length=255)
+    page_title = models.CharField(blank=True, max_length=255)
+
+    def __str__(self):
+        return self.identifier
```

Comment:
yes, thanks for the correction.
Thats what i intended but did it incorrectly.

Reply:
i take that back. i needed __str__ only so that in the views/html it's rendered like that. __repr__ didn't do that for me.

Reply:
Yeah, `__repr__` is just for poking around in the command line and similar.  You might also consider using an explicit `render` method or something if this is intended to be shown to the user, otherwise it can be hard to trace down where this is used if you wanted to change it later.

Reply:
Oh, or pass `obj.identifier` to the template when it's going to be rendered rather than `obj`

Reply:
okay. removed it and used identifier.
I was using it to populate choices for link on `CCZHostingForm`